app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt

```diff
@@ -1487,10 +1493,13 @@ class PlayerFragment : Fragment(), OnlinePlayerOptions {
                     if (currentQuality == it.resolution) "${it.name} ✓" else it.name
                 }
             ) { which ->
-                val newResolution = resolutions[which].resolution
-                setPlayerResolution(newResolution)
+                var newResolution = resolutions[which].resolution
+                setPlayerResolution(newResolution, true)
 
                 // save the selected resolution to update on fullscreen change
+                if (isShort) {
+                    newResolution = ceil(newResolution * 9.0 / 16.0).toInt()
+                }
```

Comment:
I think this code here should be removed, since we already handle that in `setPlayerResolution`?

Reply:
I don't really like/agree on this change, it's not a good idea to modify the `newResolution` variable, it should always be in the "same format" (normal horizontal video ratio resolution).

Reply:
This here makes everything more complicated and I don't think anybody knows what's going on when maintaining that code in the future.